packages/multichain-network-controller/src/MultichainNetworkController.ts

```diff
@@ -137,6 +139,52 @@ export class MultichainNetworkController extends BaseController<
     return await this.#setActiveEvmNetwork(id);
   }
 
+  /**
+   * Removes an EVM network from the list of networks.
+   * This method re-directs the request to the network-controller.
+   *
+   * @param chainId - The chain ID of the network to remove.
+   * @returns - A promise that resolves when the network is removed.
+   */
+  async #removeEvmNetwork(chainId: CaipChainId): Promise<void> {
+    const hexChainId = convertCaipToHexChainId(chainId);
+    const selectedChainId = this.messagingSystem.call(
+      'NetworkController:getSelectedChainId',
+    );
+
+    if (this.state.isEvmSelected && selectedChainId === hexChainId) {
+      throw new Error('Cannot remove the currently selected network');
+    }
+
+    if (!this.state.isEvmSelected && selectedChainId === hexChainId) {
+      const ethereumMainnetHexChainId = '0x1';
+      const clientId = this.messagingSystem.call(
+        'NetworkController:findNetworkClientIdByChainId',
+        ethereumMainnetHexChainId,
+      );
+
+      await this.messagingSystem.call(
+        'NetworkController:setActiveNetwork',
+        clientId,
+      );
+    }
```

Comment:
Cosmetic: Just another alternative with some more comments:

```suggestion
    if (selectedChainId === hexChainId) {
      // We prevent removing the currently selected network.
      if (this.state.isEvmSelected) {
        throw new Error('Cannot remove the currently selected network');
      }

      // If a non-EVM network is selected, we can delete the currently EVM selected network, but
      // we automatically switch to EVM mainnet.
      const ethereumMainnetHexChainId = '0x1'; // TODO: Should probably be a constant.
      const clientId = this.messagingSystem.call(
        'NetworkController:findNetworkClientIdByChainId',
        ethereumMainnetHexChainId,
      );

      await this.messagingSystem.call(
        'NetworkController:setActiveNetwork',
        clientId,
      );
    }
```